plenum/server/request_managers/write_request_manager.py

```diff
@@ -302,59 +308,43 @@ def do_taa_validation(self, request: Request, req_pp_time: int, config):
                 )
                 return
 
+        if not self.get_taa_data():
+            logger.trace(
+                "{} TAA acceptance passed for request {}: "
+                "taa is disabled".format(self, request.reqId)
+            )
+            return
+
+        if not request.taaAcceptance:
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "Txn Author Agreement acceptance is required for ledger with id {}"
+                .format(ledger_id)
+            )
+
         taa = None
-        taa_data = self.get_taa_data()
+        r_taa_a_digest = request.taaAcceptance.get(f.TAA_ACCEPTANCE_DIGEST.nm)
+        taa_data = self.get_taa_data(digest=r_taa_a_digest)
         if taa_data is not None:
             (taa, taa_seq_no, taa_txn_time), taa_digest = taa_data
-
-        if taa is None:
-            if request.taaAcceptance:
-                raise InvalidClientTaaAcceptanceError(
-                    request.identifier, request.reqId,
-                    "Txn Author Agreement acceptance has not been set yet"
-                    " and not allowed in requests"
-                )
-            else:
-                logger.trace(
-                    "{} TAA acceptance passed for request {}: taa is not set"
-                    .format(self, request.reqId)
-                )
-                return
-
-        if not taa[TXN_AUTHOR_AGREEMENT_TEXT]:
-            if request.taaAcceptance:
-                raise InvalidClientTaaAcceptanceError(
-                    request.identifier, request.reqId,
-                    "Txn Author Agreement acceptance is disabled"
-                    " and not allowed in requests"
-                )
-            else:
-                logger.trace(
-                    "{} TAA acceptance passed for request {}: taa is disabled"
-                    .format(self, request.reqId)
-                )
-                return
+        else:
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "Incorrect Txn Author Agreement(digest={}) in the request".format(r_taa_a_digest)
+            )
 
         if not taa_digest:
             raise LogicError(
                 "Txn Author Agreement digest is not defined: version {}, seq_no {}, txn_time {}"
```

Comment:
Do we still need this check?

Reply:
Because it's a strange check with LogicError (by Andrew). I didn't see a reason in this before changes and now too.
I can remove it.